main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -237,7 +237,7 @@ object Namer extends Phase[WeededAst.Program, NamedAst.Root] {
             prog0.copy(properties = prog0.properties + (ns0 -> (property :: properties)))
         }
 
-      case WeededAst.Declaration.Sig(doc, ann, mod, ident, tparams, fparams, tpe, eff, loc) =>
+      case _: WeededAst.Declaration.Sig =>
         throw InternalCompilerException("Unexpected signature declaration.") // signatures should not be at the top level
```

Comment:
The grammar ensures that this cannot happen, right?

Reply:
Right.